BN_pseudo_rand is really BN_rand

richsalz opened the pull request:
And BN_pseudo_rand_range is really BN_rand_range

I am starting to look at updating the RAND family for this release, and I started looking at bn_rand. It turns out that the "psuedo" BN rand functions don't do anything different.  So I renamed a parameter to be more clear, and replaced BN_pseudo_rand{_range} with BN_rand{_range}

richsalz commented:
Added a commit which deprecates them.

levitte commented on include/openssl/bn.h:
```diff
@@ -154,9 +154,10 @@ void BN_CTX_start(BN_CTX *ctx);
-int BN_pseudo_rand_range(BIGNUM *rnd, const BIGNUM *range);
+DEPRECATEDIN_1_1_0(int BN_pseudo_rand(BIGNUM *rnd,
+                                      int bits, int top, int bottom))
+DEPRECATEDIN_1_1_0(int BN_pseudo_rand_range(BIGNUM *rnd, const BIGNUM *range))
```
Not ok, this is a change of an existing API.  You may, however, use `DEPRECATEDIN_1_2_0`, it just needs to be defined in `include/openssl/opensslconf.h.in` (I'm surprised, I thought that was already added...  by @mattcaswell if I remember correctly)

mattcaswell replied:
> (I'm surprised, I thought that was already added... by @mattcaswell if I remember correctly)

It was in #3544 which never got merged.

levitte replied:
Thanks.  So @richsalz, this is the commit to snatch: 2597a2eb44f18474b0c82338ea80e46b3b333c07

levitte commented on crypto/bn/bn_rand.c:
```diff
@@ -174,14 +172,9 @@ static int bn_rand_range(int pseudo, BIGNUM *r, const BIGNUM *range)
-int BN_rand_range(BIGNUM *r, const BIGNUM *range)
-{
-    return bn_rand_range(0, r, range);
-}
```
Errr, so where is it implemented?  It's still declared in `bn.h`!

levitte commented on crypto/bn/bn_rand.c:
```diff
@@ -174,14 +172,9 @@ static int bn_rand_range(int pseudo, BIGNUM *r, const BIGNUM *range)
-    return bn_rand_range(0, r, range);
-}
-
 int BN_pseudo_rand_range(BIGNUM *r, const BIGNUM *range)
```
This function needs to be guarded with:

``` C
#if (OPENSSL_API_COMPAT < 0x10200000L)
```

levitte commented on crypto/bn/bn_rand.c:
```diff
@@ -100,19 +100,17 @@ int BN_rand(BIGNUM *rnd, int bits, int top, int bottom)
 
 int BN_pseudo_rand(BIGNUM *rnd, int bits, int top, int bottom)
```
This function needs to be guarded with:

``` C
#if (OPENSSL_API_COMPAT < 0x10200000L)
```

richsalz commented:
updated the second commit to do the deprecation right for 1.2.0
The old static ```bn_rand_range``` was renamed to ```BN_rand_range```
Addressed all issues so far.

kroeckx commented on include/openssl/opensslconf.h.in:
```diff
@@ -95,6 +95,12 @@ extern "C" {
+# define DEPRECATEDIN_1_2_0(f)   DECLARE_DEPRECATED(f)
+#else
+# define DEPRECATEDIN_1_2_0(f)
+#endif
```
I seem to be missing something. We already have variables marked as deprecated in 1.2.0, but I can't find how.

mattcaswell replied:
Using OPENSSL_API_COMPAT, e.g.

````
# if OPENSSL_API_COMPAT < 0x10200000L
/*
 * LONG and ZLONG are strongly discouraged for use as stored data, as the
 * underlying C type (long) differs in size depending on the architecture.
 * They are designed with 32-bit longs in mind.
 */
DECLARE_ASN1_ITEM(LONG)
DECLARE_ASN1_ITEM(ZLONG)
# endif
````

richsalz commented:
There was a PR but it’s not merged yet.  Is that what you meant?

richsalz commented:
never mind prev comment

richsalz commented:
ping.  we shouldn't be calling deprecated functions inside the library :)

richsalz commented:
I changed the documentation to say “may be deprecated” and remove the “1.2.0 deprecation” stuff. I think the builds will pass now, let’s see.

richsalz commented:
Build failures are unrelated.  I will to nudge them again.

levitte commented:
A rebase on a fresher master might help

richsalz commented:
Rebased and pushed.

richsalz commented:
builds clean, ping @mattcaswell

richsalz commented:
thanks.
